pkg/anago/anago.go

```diff
@@ -285,12 +305,13 @@ func (r *Release) Run() error {
 	}
 
 	logrus.Info("Pushing git objects")
-	if err := r.client.PushGitObjects(); err != nil {
+	// FIXME: These args, from where ??
```

Comment:
I'm wondering if we should have all  data from the release process accesible from the options struct, or even in a property in the implementation instead of passing different versions of the same data to each method. 

We currently have methods that get `*release.Versions`, `versions []string` and now I need a list of the tags in strings. Perhaps we could have a new property, like:

```go
type DefaultRelease struct {
	impl    releaseImpl
	options *ReleaseOptions
        currentRelease  XXX
}
```
An object that we could query, like: `currentRelease.Tags()`, `currentRelease.SemVer()`, `currentRelease.Versions()` and so on. 

Like, keeping the parameters in `options` and all calculated/inferred/environment data in the new object.

Reply:
Yes this makes sense to me. I'm not happy with all those args passed around so I think it's better to crunch them together into one type. We should not mix up terminology: the `build-version` is the jenkins build version, containing a commit/build. `versions []string` aka `release.Versions.Ordered()` are the versions to be built. We could go more abstract and define something like a `State struct`, which then can be modified during the whole release process and will act as our shared mutable state.

Reply:
This is done in #1726  :heavy_check_mark: